src/client/pages/Offer/Introduction/Sidebar/index.tsx

```diff
@@ -0,0 +1,280 @@
+    flex-shrink: 0;
+  }
+`
+
+const Container = styled.div`
+  width: ${SIDEBAR_WIDTH};
+  max-width: 100%;
+  padding: 1.25rem 1rem 1rem 1rem;
+  margin-top: 3rem;
+  background-color: ${colorsV3.white};
+  border-radius: 8px;
+  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.08);
+
+  ${LARGE_SCREEN_MEDIA_QUERY} {
+    position: fixed;
+    top: 8rem;
+    margin-top: 0;
+  }
+`
+
+const DiscountInfo = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  flex-wrap: wrap;
+  align-items: flex-start;
+  margin-bottom: 0.5rem;
+`
+
+const DiscountBadge = styled(Badge)`
+  &:not(:last-child) {
+    margin-right: 0.5rem;
+    margin-bottom: 0.5rem;
+  }
+`
+
+const Header = styled.div`
+  position: relative;
+  width: 100%;
+  display: flex;
+  align-items: baseline;
+  justify-content: space-between;
+  margin-bottom: 1.5rem;
+  flex-wrap: wrap;
+
+  ${SMALL_SCREEN_MEDIA_QUERY} {
+    flex-wrap: nowrap;
+  }
+`
+
+const Title = styled.h3`
+  width: 100%;
+  margin: 0 1rem 0 0;
+  font-size: 1.375rem;
+  line-height: 1.625rem;
+  white-space: pre-line;
+  color: ${colorsV3.gray900};
+
+  ${LARGE_SCREEN_MEDIA_QUERY} {
+    font-size: 1.5rem;
+    line-height: 2rem;
+  }
+`
+
+const Body = styled.div`
+  margin-bottom: 2.5rem;
+  font-size: 0.875rem;
+  line-height: 1.25rem;
+`
+
+const BodyTitle = styled.div`
+  margin-bottom: 0.75rem;
+  color: ${colorsV3.gray900};
+`
+
+const Footer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 100%;
+`
+
+const FooterExtraActions = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  margin-top: 0.75rem;
+  margin-bottom: 0.25rem;
+`
+
+export type SidebarProps = {
+  offerData: OfferData
+  campaign: CampaignDataFragment | null
+  refetchOfferData: () => Promise<void>
+  onCheckoutOpen: () => void
+}
+
+export const Sidebar: React.FC<SidebarProps> = ({
+  offerData,
+  campaign,
+  refetchOfferData,
+  onCheckoutOpen,
+}) => {
+  const { id: quoteCartId } = useParams<{ id: string }>()
+  const textKeys = useTextKeys()
+
+  const [campaignCodeModalIsOpen, setCampaignCodeModalIsOpen] = useState(false)
+  const [isSidebarVisible, setIsSidebarVisible] = useState(true)
+
+  const [addCampaignCode] = useAddCampaignCodeMutation({
+    refetchQueries: ['QuoteCart'],
+    awaitRefetchQueries: true,
+  })
+  const [removeCampaignCode] = useRemoveCampaignCodeMutation({
+    refetchQueries: ['QuoteCart'],
+    awaitRefetchQueries: true,
+  })
+
+  const campaignText = campaign?.displayValue ?? ''
+
+  const handleAddCampaignCode = useCallback(
+    async (code: string) => {
+      const { data: result } = await addCampaignCode({
+        variables: { id: quoteCartId, code },
+      })
+      const hasError = result?.quoteCart_addCampaign.__typename === 'BasicError'
+
+      if (!hasError) {
+        CampaignCode.save(code)
+      }
+
+      return { hasError }
```

Comment:
Should we handle exceptions here also?

Reply:
The thing is I'm sure what to do with them here: I believe return `false` would not be enough. Should I log it with Sentry? Feels like a scenario for ErrorBoundry and generic toast messages 🤔

Reply:
Yep agreed, as long as we catch the exception somewhere. But we can merge and test!